modules/collect/src/main/java/com/opengamma/collect/result/Result.java

```diff
@@ -450,7 +454,8 @@ public boolean isFailure() {
 
   //-------------------------------------------------------------------------
   /**
-   * Returns the actual result value if calculated successfully.
+   * Returns the actual result value if calculated successfully, throwing an
+   * exception if a failure occurred.
    * <p>
    * If this result is a failure then an an IllegalStateException will be thrown.
```

Comment:
Add a note to say that it's idiomatic to use ``map`` or ``flatMap`` instead of directly calling ``getValue``